azurerm/internal/services/azurestackhci/validate/hyper_converged_cluster.go

```diff
@@ -0,0 +1,37 @@
+package validate
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/azurestackhci/parse"
+)
+
+func HyperConvergedClusterID(i interface{}, k string) (warnings []string, errors []error) {
+	v, ok := i.(string)
+	if !ok {
+		errors = append(errors, fmt.Errorf("expected type of %q to be string", k))
+		return
+	}
+
+	if _, err := parse.HyperConvergedClusterID(v); err != nil {
+		errors = append(errors, fmt.Errorf("Can not parse %q as a resource id: %v", k, err))
+		return
+	}
+
+	return warnings, errors
+}
+
+func HyperConvergedClusterName(i interface{}, k string) (warnings []string, errors []error) {
+	v, ok := i.(string)
+	if !ok {
+		errors = append(errors, fmt.Errorf("expected type of %q to be string", k))
+		return
+	}
+
+	if matched := regexp.MustCompile(`^.{1,260}$`).Match([]byte(v)); !matched {
+		errors = append(errors, fmt.Errorf("%q may only be up to 260 characters in length", k))
+	}
```

Comment:
It is hard to believe that the name of this resource has only one constraint on the length but does not have any constraint on the characters allowed.
If this is the only constraint we know for now, could we switch the regular expression to directly check the length for better readability? 
Also, it is the best if we could have the full and accurate constraint for this name - including the length constraint and any invalid characters - for instance, I believe this name that only have several spaces is clearly illegal

Reply:
Sorry. I didn't find any constraint rule from azure doc and error message except length constraint. So I have to use length validation.